src/field_theory/splitting_field.lean

```diff
@@ -12,31 +10,64 @@ import ring_theory.polynomial
 import field_theory.minpoly
 import linear_algebra.finite_dimensional
 import tactic.field_simp
+import algebra.polynomial.big_operators
+
+/-!
+# Splitting fields
+
+This file introduces the notion of a splitting field of a polynomial and provides an embedding from
+a splitting field to any field that splits the polynomial. A polynomial `f : polynomial K` splits
+over a field extension `L` of `K` if it is zero or all of its irreducible factors over `L` have
+degree `1`. A field extension of `K` of a polynomial `f : polynomial K` is called a splitting field
+if it is the smallest field extension of `K` such that `f` splits.
+
+## Main definitions
+
+* `polynomial.splits i f`: A predicate on a field homomorphism `f : K → L` and a polynomial saying
+  that `f` is zero or all of its irreducible factors over `L` have degree `1`.
+* `polynomial.splitting_field f`: A fixed splitting field of the polynomial `f`.
+* `polynomial.is_splitting_field`: A predicate on a field to be a splitting field of a polynomial
+  `f`.
+
+## Main statements
+
+* `polynomial.C_leading_coeff_mul_prod_multiset_X_sub_C`: If a polynomial has as many roots as its
+  degree, it can be written as the product of its leading coefficient with `∏ (X - a)` where `a`
+  ranges through its roots.
+* `lift_of_splits`: If `K` and `L` are field extensions of a field `F` and for some finite subset
+  `S` of `K`, the minimal polynomial of every `x ∈ K` splits as a polynomial with coefficients in
+  `L`, then `algebra.adjoin F S` embeds into `L`.
+* `polynomial.is_splitting_field.lift`: An embedding of a splitting field of the polynomial `f` into
+  another field such that `f` splits.
+* `polynomial.is_splitting_field.alg_equiv`: Every splitting field of a polynomial `f` is isomorpic
+  to `splitting_field f` and thus, being a splitting field is unique up to isomorphism.
+
+-/
 
 noncomputable theory
 open_locale classical big_operators
 
 universes u v w
 
-variables {α : Type u} {β : Type v} {γ : Type w}
+variables {K : Type u} {L : Type v} {F : Type w}
```

Comment:
To my ears, F < K < L sounds better than K < L < F. Is there a reason for the ordering that you chose?

Reply:
I looked at the wikipedia article on splitting fields and it used L a splitting field of a polynomial over K. I then looked for one other reasonable letter for a field and took F.

Reply:
I see. K < L is a reasonable choice for the first half. When defining the `is_scalar_tower` later on, would it make sense to have F be the bottom of the `is_scalar_tower`, rather than the top?

Reply:
That sounds reasonable, I'll take a look later.

Reply:
I tried to make it work, but wasn't able to change the order mid-file. Feel free to give it a try yourself if you want. I might have done something wrong.

Reply:
I took a stab at reordering the variables in the is_scalar_tower section. Hopefully it compiles fine.

Reply:
Thanks